### ISSUE TITLE
Dismiss menus/dialogs with [Escape] AND graphical "X" (Close)

### ISSUE THREAD

**Issue by MattAtDor** [Office-of-Digital-Services/California-State-Web-Template-Website#1638]
Considering accessibility, when navigating the site with a keyboard only, popup dialogs should be able to be closed with the escape key, not just by focusing on the "X" (Close button) and then activating it with [Enter] or [Space].  
By allowing the [Escape] key to dismiss a dialog box/menu this will create an equivalent of "clicking out of the dialog box" that a mouse user can accomplish and make navigation more equitable for keyboard-only users.

**Comment by kkoryaka** [comment]
@MattAtDor could you please provide example of the component with the issue? Are you talking about modal component? https://template.webstandards.ca.gov/components/modal.html

**Comment by hdhillon88** [comment]
@kkoryaka  they saw it with the Hamburg menu or the mobile navigation part

**Comment by MattAtDor** [comment]
Yes, in "mobile view" or when zoomed in on PC, the Hamburger Menu being open was the discussion with being able to dismiss with [Esc]

Matthew Morgan
Department of Rehabilitation
Information Technology Services Division
Quality Assurance and Testing Services | Web Solution Services
Phone: (916) 322-5851

Submit an incident or service request in the IT Service Center<https://cador.service-now.com/sp>.

[Dept of Rehabilitation Logo]

CONFIDENTIALITY NOTICE:  This communication with its contents may contain confidential and/or legally privileged information.  It is solely for the use of the intended recipient(s).  Unauthorized interception, review, use or disclosure is prohibited and may violate applicable laws including the Electronic Communications Privacy Act.  If you are not the intended recipient, please contact the sender and destroy all copies of the communication.

**Comment by kkoryaka** [comment]
@extinah this is now updated in dev, please review (added escape and tatting out functionality)

**Comment by extinah** [comment]
Tested in mobile and in a browser. Escape and tabbing works.

**Comment by kkoryaka** [comment]
@MattAtDor we just released [v6.3.1 release](https://github.com/Office-of-Digital-Services/California-State-Web-Template-HTML/releases) that should have addressed this issue.
